src/util/date-input-helper.ts

```diff
@@ -1,34 +1,128 @@
 import dayjs from "dayjs";
+import isBetween from "dayjs/plugin/isBetween";
+import { INVALID_VALUE } from "../date-input/stand-alone-input";
+import { ChangeValueTypes, RawInputValues } from "../date-input";
+
+dayjs.extend(isBetween);
 
 export namespace DateInputHelper {
-    export const validate = (beginDate: string, afterDate: string): boolean => {
+    export const validate = (
+        beginDate: string,
+        afterDate: string,
+        disabledDates?: string[] | undefined,
+        between?: string[] | undefined
+    ): boolean => {
         let pass = false;
 
-        if (!beginDate.length || !afterDate.length) return false;
+        if (!beginDate?.length || !afterDate?.length) return false;
 
-        if (dayjs(beginDate).isAfter(afterDate)) {
-            pass = false;
+        if (dayjs(beginDate).isBefore(afterDate)) {
+            pass = true;
+        }
 
-            return pass;
+        if (dayjs(afterDate).isAfter(beginDate)) {
+            pass = true;
+        }
+
+        if (disabledDates && disabledDates.length) {
+            pass = !disabledDates.some((value) =>
+                [beginDate, afterDate].includes(value)
+            );
+
+            if (!pass) return false;
         }
 
-        if (dayjs(afterDate).isBefore(beginDate)) {
+        if (between && between.length) {
+            const result = [beginDate, afterDate].map((selectedDate) =>
+                dayjs(selectedDate).isBetween(
+                    between[0],
+                    between[1],
+                    "day",
+                    "[]"
+                )
+            );
+
+            pass = result.every(Boolean);
+
+            if (!pass) return false;
+        }
+
+        if (dayjs(beginDate).isAfter(afterDate)) {
             pass = false;
 
             return pass;
         }
 
-        if (dayjs(beginDate).isBefore(afterDate)) {
-            pass = true;
+        return pass;
```

Comment:
yes. it will straight return the value if the value is false. 

Only return true at the end once the both value pass in the first condition and skip other below condition.

```
if (dayjs(startDate).isBefore(endDate)) {
      pass = true;
}
``

Reply:
removed the `pass` variable.  updated to:

```
() => {
 if(cond1) {
    return false
  }

  if(cond2) {
    return false
  }

 return true
}

```